barustenberg/src/srs/reference_string/mem_reference_string.rs

```diff
@@ -1,17 +1,25 @@
 use ark_bn254::G2Affine;
+use ark_serialize::CanonicalDeserialize;
 
 use super::VerifierReferenceString;
 
-#[derive(Debug, Default)]
+#[derive(Debug)]
 pub(crate) struct VerifierMemReferenceString {
     g2_x: G2Affine,
 }
 
 impl VerifierMemReferenceString {
     pub(crate) fn new(_g2x: &[u8]) -> Self {
-        // Add the necessary code to convert g2x bytes into g2::AffineElement
-        // and initialize precomputed_g2_lines
-        unimplemented!()
+        let g2_x = match G2Affine::deserialize_uncompressed(&*_g2x) {
```

Comment:
@laudiacay do you know if we're using uncompressed affline elements or compressed?

Reply:
uhhh grep around in the codebase but i believe i've uniformly used uncompressed! there are a lot of usages in transcript

Reply:
@laudiacay cool. Out of office this week but will sort it Saturday or Monday

Reply:
Yep - this is good to go. added a `from_affline` as well